daemon/main.go

```diff
@@ -138,6 +138,11 @@ func init() {
 						Name:        "keep-config",
 						Usage:       "When restoring state, keeps containers' configuration in place",
 					},
+					cli.BoolFlag{
+						Destination: &config.KVStoreIPv4Registration,
```

Comment:
I'm sure the KV store supports an atomic lookup-or-add operation which allows to uniquely register the address or receive the information that somebody else is using it.